src/workerd/api/node/crypto.h

```diff
@@ -55,9 +56,33 @@ class CryptoImpl final: public jsg::Object {
   // Primes
   kj::Array<kj::byte> randomPrime(uint32_t size, bool safe,
       jsg::Optional<kj::Array<kj::byte>> add, jsg::Optional<kj::Array<kj::byte>> rem);
-
   bool checkPrimeSync(kj::Array<kj::byte> bufferView, uint32_t num_checks);
 
+  // Hash
+  class HashHandle final: public jsg::Object {
+    public:
+      HashHandle(kj::String& algorithm, kj::Maybe<uint32_t> xofLen);
+      HashHandle(EVP_MD_CTX* in_ctx, kj::Maybe<uint32_t> xofLen);
+      ~HashHandle();
+
+      jsg::Ref<HashHandle> copy(jsg::Lock& js, kj::Maybe<uint32_t> xofLen);
+      int update(jsg::Lock& js, kj::Array<kj::byte> data);
+      kj::Array<kj::byte> digest(jsg::Lock& js);
+      static jsg::Ref<HashHandle> constructor(jsg::Lock& js, kj::String algorithm, kj::Maybe<uint32_t> xofLen);
+
+      JSG_RESOURCE_TYPE(HashHandle) {
+        JSG_METHOD(update);
+        JSG_METHOD(digest);
+        JSG_METHOD(copy);
+      };
+
+    private:
+      void checkDigestLength(const EVP_MD* md, kj::Maybe<uint32_t> xofLen);
+
+      EVP_MD_CTX* md_ctx;
```

Comment:
Recommend wrapping this in a `kj::Own`... see uses of `OSSL_NEW(EVP_MD_CTX)` in, for instance, crypto-impl-asymmetric.c++ for an example.